.github/workflows/livestream-docker-image.yml

```diff
@@ -0,0 +1,84 @@
+name: Docker CI/CD
+
+on:
+    push:
+        branches:
+            - master
+        push:
+            branches:
+                - master
+        paths:
+            - 'livestream/**'
+
+jobs:
+    build:
+        runs-on: ubuntu-latest
+
+        permissions:
+            contents: read
+            packages: write
+
+        outputs:
+            sha: ${{ steps.push.outputs.digest }}
+
+        steps:
+            - uses: actions/checkout@v4
+            - name: Log in to the Container registry
+              uses: docker/login-action@v3
+              with:
```

Comment:
in rust, I only checkout the livestream folder, which makes it very clear when I'm using the wrong dockerfile for example